lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/FeatureService.java

```diff
@@ -529,4 +534,149 @@ public List<Feature> readFeaturesFromFeatureListFile(List<Feature> installedFeat
         }
         return installedFeatures;
     }
+
+    /**
+     * get all platforms for all features from feature json
+     * @param libertyVersion liberty version
+     * @param libertyRuntime runtime
+     * @param requestDelay request delay
+     * @param documentURI document uri link
+     * @return set of unique platforms
+     */
+    public Set<String> getAllPlatforms(String libertyVersion, String libertyRuntime, int requestDelay, String documentURI) {
+        List<Feature> features = this.getFeatures(libertyVersion, libertyRuntime, requestDelay, documentURI);
+        return features.stream()
+                .map(Feature::getWlpInformation)
+                .filter(Objects::nonNull)
+                .map(WlpInformation::getPlatforms)
+                .filter(Objects::nonNull)
+                .flatMap(List::stream).collect(Collectors.toSet());
+    }
```

Comment:
Should we cache the platforms like we cache the features in `getFeatures`? As long as the runtime/version have not changed, the list of platforms should not change. No need to do this logic over and over.

Reply:
No need to cache platform separately. If you see above, we are calling getFeatures() method, which already returns cached features. platform is also cached with feature data

Reply:
True, but we are collecting something over and over that we already know after the first time. Guess we can optimize later if we observe that it is slow.